src/raw/mod.rs

```diff
@@ -880,6 +957,7 @@ impl<'f, 'a, A: Automaton> Streamer<'a> for Stream<'f, A> {
             let trans = state.node.transition(state.trans);
             let out = state.out.cat(trans.out);
             let next_state = self.aut.accept(&state.aut_state, trans.inp);
+            let aut_state = self.aut.accept(&state.aut_state, trans.inp);
```

Comment:
I am asking for the state a second because the `Automaton` `State` needs to be returned from this function and saved into the `stack`, so the other solution would be to ask a `Clone`able `Automaton` `State` but it add to many restriction to the entire code base (`Clone` and `Clone` constraints everywhere).

The solution would be to ask for a `Clone`able `State` **only** for the `SreamWithState` and in that case not be able to reuse the `StreamWithState` in the `Stream`, to add `Clone` constraints everywhere or the current solution choose: ask two times for the state.